src/services/coursesService.js

```diff
@@ -1,16 +1,27 @@
-const courses = require('../databases/coursesDb')
+const courses = require('../databases/coursesDb');
 
-const getCourses = ({page}) => {
-  return courses.getCourses({page, limit});
-}
+const getCourse = async ({ id }) => courses.getCourse({ id });
 
-const addCourse = ({description, name}) => {
+const getCourses = async ({ page, limit, userToken }) => {
+  // TODO: users service integration for the moment we use the token
+  const userId = userToken;
+  const coursesByUser = await courses.getCoursesByUser({ page, limit, userId });
+  const result = coursesByUser.map((c) => getCourse({ id: c.courseId }));
```

Comment:
you are missing an `await`. Also that approach is not correct, you should be able to list all the courses in the upper line (`getCoursesByUser`)

Reply:
you should use a join in `getCoursesByUser`. Let's talk about it